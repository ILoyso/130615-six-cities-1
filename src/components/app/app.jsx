import React from 'react';
import PropTypes from 'prop-types';

import MainScreen from '../main-screen/main-screen.jsx';


/**
 * Application component, here the whole process begins
 * @param {Object} props
 * @return {*}
 */
const App = (props) => {
  const {places} = props;

  return <MainScreen
    places={places}
  />;
};


App.propTypes = {
  places: PropTypes.arrayOf(PropTypes.shape({
    title: PropTypes.string.isRequired,
    img: PropTypes.string.isRequired,
    type: PropTypes.string.isRequired,
    price: PropTypes.number.isRequired,
    rating: PropTypes.number.isRequired,
    isPremium: PropTypes.bool.isRequired,
  })).isRequired
};


export default App;
