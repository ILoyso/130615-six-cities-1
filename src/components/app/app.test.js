import React from 'react';
import Enzyme, {shallow} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import {shallowToJson} from 'enzyme-to-json';

Enzyme.configure({adapter: new Adapter()});

import {App} from './app.jsx';

describe(`App Component`, () => {
  it(`App correctly renders`, () => {
    const app = shallow(<App
      places={[
        {id: 1},
        {id: 2}
      ]}
    />);

    expect(shallowToJson(app)).toMatchSnapshot();
  });
});
