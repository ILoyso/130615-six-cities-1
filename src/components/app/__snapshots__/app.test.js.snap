// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Component App correctly renders 1`] = `
<Fragment>
  <ConnectFunction />
  <Switch>
    <Route
      exact={true}
      path="/"
      render={[Function]}
    />
    <Route
      path="/login"
      render={[Function]}
    />
    <Route
      path="/favorites"
      render={[Function]}
    />
    <Route
      key="0"
      path="/offer/1"
      render={[Function]}
    />
    <Route
      key="1"
      path="/offer/2"
      render={[Function]}
    />
  </Switch>
</Fragment>
`;
